Tensile/Components/LocalRead.py

```diff
@@ -0,0 +1,269 @@
+                        highVgpr = vgpr(tmpVgprIdx)
+                        destVgpr = highVgpr
+                    if isHigh8Bits:
+                        lowVgpr = vgpr(tmpVgprIdx-1) if isHigh16Bits else baseLRVgpr
+                        packCode.addInst("_v_lshl_or_b32", lowVgpr, highVgpr, "0x8", lowVgpr, "pack two int8 Vgpr to one half Vgpr")
+                        if isHigh16Bits:
+                            packCode.addInst("v_or_b32", baseLRVgpr, baseLRVgpr, lowVgpr, "pack two half Vgpr to one Vgpr")
+                    if isHigh8Bits or isHigh16Bits:
+                        tmpVgprIdx += 1
+
+                valufIdx += blockWidth
+
+                # load read instrution
+                paramList = []
+                paramList.append(destVgpr)
+                paramList.append(vgpr("LocalReadAddr%s"%tc))
+
+                for oIdx in range(0, numOffsets):
+                    offset_val = (vIdx * numOffsets+oIdx) * MIWaveGropuShape[tIdx] * tileStride
+                    offset_val = (rIdx * UnrollStride + offset_val + tP["localReadOffset"]) * tP["bpe"]
+                    if (kernel["LdsBlockSizePerPad%s"%tc] != 0) and (kernel["LdsPad%s"%tc] != 0):
+                        offset_val = offset_val + (offset_val // kernel["LdsBlockSizePerPad%s"%tc]) * kernel["LdsPad%s"%tc] * tP["bpe"]
+                    offset_val = offset_val + tP["localReadSwapByteOffset"]
+                    paramList.append(int(offset_val))
+
+                paramTuple = tuple(paramList)
+                comment = "L -> Reg lro=%d swapByteOffset=%u ti=%u vIdx=%u rIdx=%u oIdx=%u buffer=%u iui=%u" \
+                        % (tP["localReadOffset"], tP["localReadSwapByteOffset"], MIWaveGropuShape[tIdx], vIdx, rIdx, oIdx, bufferIdx, iui)
+
+                highBits = highBitsForHalf or isHigh16Bits
+                localReadCode.addCode(Code.LocalReadInst(instruction.IssueLatency,instruction.toCodeInst(paramTuple, 0, highBits), comment))
+
+                # TODO - handle vector-load
+                tmpSgpr = writer.getTmpSgpr(1).idx()
+                if writer.db["CheckValue1%s"%tc] and not writer.inTailLoop:
+
+                    dbgVgpr = destVgpr
+                    dbgVgprList = destVgpr.split("v[")
+                    if len(dbgVgprList) == 1: # vIdx, no []
+                        dbgVgpr = dbgVgprList[0]
+                    else:
+                        # We only check the first one now
+                        # TODO: Handle vector, but need to take care the last one
+                        dbgVgprList = (dbgVgprList[1].split("]")[0]).split(':')
+                        dbgVgpr = "v[%s]"%dbgVgprList[0]
+
+                    localReadCode.addInst("s_waitcnt lgkmcnt(0)", "CheckValue1 wait for LDS read")
+                    if writer.archCaps["SeparateVscnt"]:
+                        localReadCode.addInst( "s_waitcnt_vscnt", "null", "0", "")
+
+                    if kernel["ProblemType"]["DataType"].isHalf():
```

Comment:
Could be a candidate for another component, or maybe just adding literal value generation to the data type code?